### ISSUE TITLE
Make Location Optional in Formation

### ISSUE THREAD

**Issue by hinzed1127** [hinzed1127/pivotal-to-gh-importer#430]
This issue was imported from Pivotal Tracker.

  | Field | Value |
  | --- | --- |
  | Story ID | [#184125110](https://www.pivotaltracker.com/story/show/184125110) |
  | State | accepted |
  | Requested By | dillonsullivan |
  | Owned By | aarondidner |
  | Estimate | 1 |
  | Priority | p3 - Low |
  | Created At | Dec 30, 2022 |
  | Accepted At | Mar 21, 2023 |
      
## User Stories

As a Navigator using, I want not to have to provide my location information during business formation so that I have less barriers towards starting my business. 

## Acceptance Criteria

- Given I am Poppy on the formation task
when on the step 2
  - then the location fields are no longer required
  - the optional header is added
  - the field begins collapsed as the others that are optional on the page
  - the submit error handling logic is updated to make sure either all the fields are blank or filled out and not partially filled out. 

- Given I am Dakota on the formation task
when on the step 2
then the location fields are required and otherwise unchanged from current look (IE not collapsing, no optional label in header)

## Assumptions

- Auto-fill address will continue to function

## Analytics

- If the location is not set, then the location dimension should not be set, right?

## ADO User Story

https://dev.azure.com/NJInnovation/Business%20First%20Stop/_workitems/edit/2355

## Comments
@jessica2I I think you and I added this during the holiday bridge and at the time didn't add details. I've since added some baseline AC to review with the team. Please review and inform me if it's off-base. (dillonsullivan - Jan 11, 2023)

(Anne LoVerso - Jan 12, 2023)

For member and registered agent auto-fill address, we need to determine if something needs to change for the best experience @jessica2I . (dillonsullivan - Jan 12, 2023)

@dillonsullivan @ej3 this will break some existing functionality from deferred location. We have logic where we always make sure someone has provided their location by the time they have finished Formation - either via the formation modal, or the formation task, or a deferred location task, etc.

By removing location as required from Formation, we will now have users who progress into the NEEDS_TO_REGISTER phase without having ever provided their location. Is this desired? If so, when do you want those users to be asked for their municipality? (Anne LoVerso - Jan 17, 2023)

@anneloverso_nj this is desired, but to your point do we need to do something else to prompt them for their location, which is required for registering for taxes @jessica2I @ceciliaramirez1 @caryeuwer ??Do we need a follow-up discovery task or story to prompt users in the register for taxes modal or prior? (dillonsullivan - Jan 17, 2023)

@anneloverso_nj @dillonsullivan Currently- If the user doesn't use our formation, and they haven't answered the location question elsewhere, we ask them the location question in the modal to get to one of the next phases. Couldn't we use similar logic. Any user, regardless if they used our formation, if they haven't answered the location question elsewhere, will be prompted to answer it in the same modal where the others get asked? I am not sure which modal it is. (jessica2I - Jan 17, 2023)

@jessica2I that's a good point, they do get asked the question in the registered-for-taxes modal. just double-checking that's okay because that's different behavior than we had previously (Anne LoVerso - Jan 17, 2023)

So for UX consistency if we're making the location fields optional, shouldn't we also make the location section collapsed like **Business Purpose** and **Additional Provisions/Articles**

And also maybe amend the text for **Address Line 2** which explicitly says its optional

@caryeuwer @fiorellagalvez @amandabeall  (aarondidner - Feb 27, 2023)

@aarondidner If the "Primary Business Address" section is now optional, I suggest collapsing it to keep it consistent with "Business Purpose/Additional Provisions..." 
What do you think @jessica2I?  cc/ @caryeuwer @amandabeall  (Fiorella Galvez - Feb 27, 2023)

Commit by adidner
https://github.com/newjersey/navigator.business.nj.gov/commit/a72943068d1ec56ef4e5a7889c8a6a7be0800658
navigator.business.nj.gov

feat: [#184125110] make location optional in formation for NJ (Anne LoVerso - Mar 14, 2023)

Commit by adidner
https://github.com/newjersey/navigator.business.nj.gov/commit/beb5dc26433cbc78db74a60c6e798519540aafa6
navigator.business.nj.gov

feat: [#184125110] make location optional in formation for NJ (Anne LoVerso - Mar 15, 2023)

Commit by adidner
https://github.com/newjersey/navigator.business.nj.gov/commit/fca50fea15774d667a27b6dfc86d06a208f9caa2
navigator.business.nj.gov

feat: [#184125110] make location optional in formation for NJ (Anne LoVerso - Mar 15, 2023)

Commit by adidner
https://github.com/newjersey/navigator.business.nj.gov/commit/519039546a680767b78bab361938d6512af2870a
navigator.business.nj.gov

feat: [#184125110] make location optional in formation for NJ (Anne LoVerso - Mar 15, 2023)

Commit by adidner
https://github.com/newjersey/navigator.business.nj.gov/commit/25d8a584eb6f72d4d2f1dd2a01ccda0d9d5a9997
navigator.business.nj.gov

feat: [#184125110] make location optional in formation for NJ (Anne LoVerso - Mar 16, 2023)

Commit by adidner
https://github.com/newjersey/navigator.business.nj.gov/commit/0a63b913fe08eae819104f8bd173b58a1ff5c058
navigator.business.nj.gov

feat: [#184125110] make location optional in formation for NJ (Anne LoVerso - Mar 16, 2023)

Commit by adidner
https://github.com/newjersey/navigator.business.nj.gov/commit/682070648bc0392ebd03fb027d3fd3dfca00e211
navigator.business.nj.gov

feat: [#184125110] make location optional in formation for NJ (Anne LoVerso - Mar 16, 2023)

Commit by adidner
https://github.com/newjersey/navigator.business.nj.gov/commit/e0e914affe26effd67bce2840f823e2436ff9847
navigator.business.nj.gov

feat: [#184125110] make location optional in formation for NJ (Anne LoVerso - Mar 16, 2023)

@dillonsullivan @marquesjackson @jessica2I this ticket is ready for review (aarondidner - Mar 16, 2023)

Waiting on registration. Should be resolved today.  (caseycruz - Mar 20, 2023)

**Product** review set to **pass** (marquesjackson - Mar 21, 2023)

Commit by semantic-release-bot
https://github.com/newjersey/navigator.business.nj.gov/commit/a0ee06e9ac1c96370814799e21127d88ebe0c59a
navigator.business.nj.gov

chore(release): 5.16.0 [skip ci]

# [5.16.0](https://github.com/newjersey/navigator.business.nj.gov/compare/v5.15.1...v5.16.0) (2023-03-21)

### Bug Fixes

* [[#184700586](https://github.com/newjersey/navigator.business.nj.gov/issues/184700586)] increase option limit for netlify relations dropdowns ([6653e92](https://github.com/newjersey/navigator.business.nj.gov/commit/6653e92fa198ed4a87aa65f11b05400e1587b691))
* [[#184718036](https://github.com/newjersey/navigator.business.nj.gov/issues/184718036)] welcome card now closes when the close button is hit ([84e9225](https://github.com/newjersey/navigator.business.nj.gov/commit/84e9225a3c2d01ed790f2260390e78f60f1ef930))
* added updated inline error validations on profile dakota nexus fields ([c2b570b](https://github.com/newjersey/navigator.business.nj.gov/commit/c2b570b466e15fb119845bf17c1b88f56059ded4))
* adding missing apt package for chrome installation ([39ee8ee](https://github.com/newjersey/navigator.business.nj.gov/commit/39ee8ee2b06b14be6d5b3260a6740a791b27e5e9))
* cms broken from mui theme and form context ([7e12d6d](https://github.com/newjersey/navigator.business.nj.gov/commit/7e12d6dfa44b80a7743a5e9483cece0dfd71efb2))
* **deps:** update aws-amplify ([0ecc825](https://github.com/newjersey/navigator.business.nj.gov/commit/0ecc82507bb50e94d26f4b2993e61fddaf77b701))
* **deps:** update aws-amplify ([410772e](https://github.com/newjersey/navigator.business.nj.gov/commit/410772ee4a2452aaf04545f7ac22cd890772a111))
* **deps:** update aws-sdk to v3.292.0 ([c7f2cb4](https://github.com/newjersey/navigator.business.nj.gov/commit/c7f2cb47f589cfbb2dd75e1e7ec92e28abefc8b8))
* **deps:** update aws-sdk to v3.294.0 ([d3c9238](https://github.com/newjersey/navigator.business.nj.gov/commit/d3c9238f29d79f1797ceb3ec09822407ff4b8447))
* **deps:** update dependency @aws-crypto/client-node to v3.2.0 ([0b2de4e](https://github.com/newjersey/navigator.business.nj.gov/commit/0b2de4ed57ee22b575cd6c536cb0250188b19d83))
* **deps:** update dependency @aws-crypto/sha256-browser to v4 ([4140466](https://github.com/newjersey/navigator.business.nj.gov/commit/4140466ac1f94eedeabb7eb1afeff73a4421c918))
* **deps:** update dependency body-parser to v1.20.2 ([b7417ba](https://github.com/newjersey/navigator.business.nj.gov/commit/b7417bad240ef0f0f9a3cba0920a64df5ada41b7))
* **deps:** update dependency focus-trap-react to v10.1.1 ([fb31a51](https://github.com/newjersey/navigator.business.nj.gov/commit/fb31a51713c88a4fdd1769c783657941654f2c52))
* **deps:** update dependency ua-parser-js to v1.0.34 ([9cf0ae9](https://github.com/newjersey/navigator.business.nj.gov/commit/9cf0ae9afbf64c41fe968dc3638e319af9b8f195))
* **deps:** update material-ui ([bb466f6](https://github.com/newjersey/navigator.business.nj.gov/commit/bb466f6a849041b61eb5a674329b2a9948b51699))
* **deps:** update material-ui ([3211baa](https://github.com/newjersey/navigator.business.nj.gov/commit/3211baa9f480d486281367cb1c51fd0ab8022d67))
* fixing the onboarding flow after the form refactor ([c55a914](https://github.com/newjersey/navigator.business.nj.gov/commit/c55a914acbccd78dda06ac70289c3ba88891af00))
* going back to 1 max worker in jest ci due to jest bug ([86a71b7](https://github.com/newjersey/navigator.business.nj.gov/commit/86a71b70817747023354dfc0933394527e4b05a6))
* removed brackets from test names for CI jest bug ([18b795e](https://github.com/newjersey/navigator.business.nj.gov/commit/18b795e2c9fe1f79f31fa7bc17d858d9c7dea6c7))
* setting chrome version to last stable build in docker image ([306862d](https://github.com/newjersey/navigator.business.nj.gov/commit/306862de903d1836e2e8080631da71c97d7b6cf6))
* spellcheck for UCEDC ([7eddf84](https://github.com/newjersey/navigator.business.nj.gov/commit/7eddf849b50fb2344ed6f54a4eda9dafdb9e76d2))
* updating General Config config ([#5392](https://github.com/newjersey/navigator.business.nj.gov/issues/5392)) ([d44ac47](https://github.com/newjersey/navigator.business.nj.gov/commit/d44ac470b4e287b356cecb08ab2bd54b7e85da24))
* updating General Config config ([#5393](https://github.com/newjersey/navigator.business.nj.gov/issues/5393)) ([2dde0ad](https://github.com/newjersey/navigator.business.nj.gov/commit/2dde0adb53d531756e4ee6820b1542585e127978))
* updating General Config config ([#5396](https://github.com/newjersey/navigator.business.nj.gov/issues/5396)) ([cd11a21](https://github.com/newjersey/navigator.business.nj.gov/commit/cd11a21b43126c20434b566dfc56247b8c814bec))
* updating onboarding inline error for dakota nexus location radio button ([9ce48ff](https://github.com/newjersey/navigator.business.nj.gov/commit/9ce48ff2ce144e1a5e30488875eb7b8c651fff07))
* updating Tasks - All business-plan ([#5426](https://github.com/newjersey/navigator.business.nj.gov/issues/5426)) ([a4a3686](https://github.com/newjersey/navigator.business.nj.gov/commit/a4a3686506bf6f0be3425cef31b6307e283073a0))
* updating Tasks - All business-plan ([#5427](https://github.com/newjersey/navigator.business.nj.gov/issues/5427)) ([7017307](https://github.com/newjersey/navigator.business.nj.gov/commit/70173079e899c5b7c5414dbd23ea02cadcf47d29))
* updating Tasks - All nonprofit-tax-exempt ([#5311](https://github.com/newjersey/navigator.business.nj.gov/issues/5311)) ([45f8b90](https://github.com/newjersey/navigator.business.nj.gov/commit/45f8b905ad863b6f514d8b0ec469008ee64a51f5))
* updating Tasks - Cannabis Priority Status cannabisPriority-1 ([#5391](https://github.com/newjersey/navigator.business.nj.gov/issues/5391)) ([4d0ddb9](https://github.com/newjersey/navigator.business.nj.gov/commit/4d0ddb9a7460237b793619388b11afdd118d03d5))
* updating Taxes Filings - All accounting-firm-renew ([#5400](https://github.com/newjersey/navigator.business.nj.gov/issues/5400)) ([10274de](https://github.com/newjersey/navigator.business.nj.gov/commit/10274de76c2c75df3f64c18f34e531adea4204ea))
* updating Taxes Filings - All accounting-firm-renew ([#5405](https://github.com/newjersey/navigator.business.nj.gov/issues/5405)) ([62c591c](https://github.com/newjersey/navigator.business.nj.gov/commit/62c591c1eb8ed7589e057a75f2642a31a4a987a5))
* updating Taxes Filings - All annual-report ([#5395](https://github.com/newjersey/navigator.business.nj.gov/issues/5395)) ([d43e0ac](https://github.com/newjersey/navigator.business.nj.gov/commit/d43e0ac748d49ce63aefa7c706d3fba978538ec3))
* updating Taxes Filings - All annual-report ([#5397](https://github.com/newjersey/navigator.business.nj.gov/issues/5397)) ([ef74d2b](https://github.com/newjersey/navigator.business.nj.gov/commit/ef74d2b1517dbc4679ef199746c05eb6f0214c5d))
* updating Taxes Filings - All cosmetology-shop-renew ([#5394](https://github.com/newjersey/navigator.business.nj.gov/issues/5394)) ([03bd461](https://github.com/newjersey/navigator.business.nj.gov/commit/03bd4619317af0a59a91ba3ce5e55be190ccf2f5))
* updating Taxes Filings - All massage-license-renew ([#5399](https://github.com/newjersey/navigator.business.nj.gov/issues/5399)) ([842fd2d](https://github.com/newjersey/navigator.business.nj.gov/commit/842fd2d84759c055707f42aafb137e4411d359c6))
* updating Taxes Filings - All pharmacy-permit-renew ([#5401](https://github.com/newjersey/navigator.business.nj.gov/issues/5401)) ([8224020](https://github.com/newjersey/navigator.business.nj.gov/commit/822402032dfa807bbaeea27340770040a4d6c1d9))

### Features

* [[#183444690](https://github.com/newjersey/navigator.business.nj.gov/issues/183444690)] add confirmation modal for profile formation date deletion ([71e5945](https://github.com/newjersey/navigator.business.nj.gov/commit/71e59455c8cbaff695d22778d2056b0e010c470e))
* [[#183444690](https://github.com/newjersey/navigator.business.nj.gov/issues/183444690)] update profile formation date field so user can delete the date ([2ee8735](https://github.com/newjersey/navigator.business.nj.gov/commit/2ee8735e855798d6a705d6b588ce406b5ed1cc0b))
* [[#184125110](https://github.com/newjersey/navigator.business.nj.gov/issues/184125110)] make location optional in formation for NJ ([e0e914a](https://github.com/newjersey/navigator.business.nj.gov/commit/e0e914affe26effd67bce2840f823e2436ff9847))
* [[#184533733](https://github.com/newjersey/navigator.business.nj.gov/issues/184533733)] share feedback & report something broken prompts live chat ([558b369](https://github.com/newjersey/navigator.business.nj.gov/commit/558b36993f3045ad73cf6e12330a58a35b6c6795))
* added new agency type to taxFilings ([2d5c52a](https://github.com/newjersey/navigator.business.nj.gov/commit/2d5c52a6aec424f927598e1c206fe37a12d05ec3)) (Anne LoVerso - Mar 21, 2023)

## Referenced Pull Requests
- https://github.com/newjersey/navigator.business.nj.gov/pull/5340
